cypress/integration/all/app/fleetdesktop.spec.ts

```diff
@@ -15,10 +15,74 @@ describe("Fleet Desktop", () => {
   describe("Fleet Desktop device user page", () => {
     beforeEach(() => {
       Cypress.session.clearAllSavedSessions();
-    });
-    it("serves device user page if url includes valid token", () => {
       cy.visit(`/device/${fakeDeviceToken}`);
+    });
+    it("renders the device user information and info modal", () => {
       cy.findByText(/my device/i).should("exist");
+      cy.getAttached(".status--online").should("exist");
+      cy.getAttached(".info-flex").within(() => {
+        cy.findByText(/operating system/i)
+          .next()
+          .contains(/ubuntu 20/i);
+      });
+      cy.getAttached(".info-grid").within(() => {
+        cy.findByText(/private ip address/i)
+          .next()
+          .findByText(/---/i)
+          .should("not.exist");
+      });
+      cy.getAttached(".device-user__action-button-container").within(() => {
+        cy.getAttached('img[alt="Host info icon"]').click();
+      });
+      cy.getAttached(".device-user-info__modal").within(() => {
+        cy.getAttached(".device-user-info__btn").click();
+      });
+    });
+    it("renders and searches the host's software", () => {
+      cy.getAttached(".react-tabs__tab-list").within(() => {
+        cy.findByText(/software/i).click();
+      });
+      let initialCount = 0;
+      cy.getAttached(".section--software").within(() => {
+        cy.getAttached(".table-container__results-count")
+          .invoke("text")
+          .then((text) => {
+            const fullText = text;
+            const pattern = /[0-9]+/g;
+            const newCount = fullText.match(pattern);
+            initialCount = parseInt(newCount[0], 10);
+            expect(initialCount).to.be.at.least(1);
+          });
+        cy.findByPlaceholderText(/search software/i).type("lib");
+        // Ensures search completes
+        cy.wait(1000); // eslint-disable-line cypress/no-unnecessary-waiting
```

Comment:
I wonder if there is another solution other than using `wait()`? My concern is over the past few months we're starting to add more and more waits back into the e2e tests, which we know can result in flakiness later on. `getAttached()` _should_ be retrying for 10 seconds, which should allow for the search results to render. But presumably that isn't happening?

Reply:
Hmm, let me toy with it!